fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -838,4 +850,50 @@ public ChecksumIndexInput openChecksumInput(final String name, final IOContext c
     Cache<Pair<Long, Integer>, CompletableFuture<byte[]>> getBlockCache() {
         return blockCache;
     }
+
+    /**
+     * Get a primary key segment index if enabled.
+     * @return index or {@code null} if not enabled
+     */
+    @Nullable
+    public LucenePrimaryKeySegmentIndex getPrimaryKeySegmentIndex() {
+        if (!primaryKeySegmentIndexEnabled) {
+            return null;
+        }
+        if (primaryKeySegmentIndex == null) {
+            final Subspace primaryKeySubspace = subspace.subspace(Tuple.from(PRIMARY_KEY_SUBSPACE));
+            primaryKeySegmentIndex = new LucenePrimaryKeySegmentIndex(this, primaryKeySubspace);
+        }
+        return primaryKeySegmentIndex;
+    }
+
+    // Map segment name to integer id.
+    // TODO: Could store this elsewhere, such as inside compound file.
+    public long primaryKeySegmentId(@Nonnull String segmentName, boolean create) throws IOException {
+        final String fileName = IndexFileNames.segmentFileName(segmentName, "", "pky");
+        FDBLuceneFileReference ref = getFDBLuceneFileReference(fileName);
```

Comment:
So the `.pky` file only exists as a means to map the segmentName to a unique number using the existing `getIncrement`?

Reply:
Correct. Would it be better to have a separate registry or to do that mapping separate from the file references?

Reply:
I don't think there would be a big win to have a second registry. I mostly just wanted to confirm that I wasn't missing something.